lib/libc/aarch64/gen/_setjmp.S

```diff
@@ -81,9 +81,7 @@ ENTRY(_longjmp)
 
 	/* Restore the stack pointer */
 	ldr	REG(8), [REG(0)], #(REG_WIDTH)
-#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
-	mov	c2, c8
-#else
+#if !defined(__CHERI_PURE_CAPABILITY__) || !defined(RTLD_SANDBOX)
```

Comment:
I find it easier to read as !(purecap && sandbox) rather than !purecap || !sandbox.

Reply:
Also, is RTLD_SANDBOX ever true for !purecap?  That is, can we simplify some of the #ifdef's to '#ifdef RTLD_SANDBOX` or `#ifndef RTLD_SANDBOX`?

Reply:
> Also, is RTLD_SANDBOX ever true for !purecap? That is, can we simplify some of the #ifdef's to `#ifdef RTLD_SANDBOX` or `#ifndef RTLD_SANDBOX`?

The purecap check is indeed redundant. I'll do a find-and-replace once the libsys is merged. (libsys renames RTLD_SANDBOX to CHERI_LIB_C18N.)

Reply:
@bsdjhb I've removed the checks for `__CHERI_PURE_CAPABILITY__` in #2200.